include/tvm/VariableVector.h

```diff
@@ -101,13 +101,27 @@ namespace tvm
     void add(const std::vector<VariablePtr>& variables);
     /** Same as add(VariablePtr), but for adding a vector of variables.*/
     void add(const VariableVector& variables);
-    /** Remove a variable from the vector.
+    /** Add a variable to the vector, if not already present and return the index
+      * of the variable in the vector whether it was added or not.
+      *
+      * \param v The variable to be added.
+      *
+      * \returns Index of variable \p v in the vector.
+      */
+    int addAndGetIndex(VariablePtr v);
```

Comment:
I took the convention of not exposing `size_t` in `tvm` to avoid the user having to do conversion. I changed to the use of `int` in `ProblemComputationData` in 2e48a737032e8255a966630a9b0940045093222c